cpp/src/decisiontree/batched-levelalgo/kernels.cuh

```diff
@@ -288,7 +352,15 @@ __global__ void computeSplitClassificationKernel(
   auto* sDone = alignPointer<int>(sbins + nbins);
   IdxT stride = blockDim.x * gridDim.x;
   IdxT tid = threadIdx.x + blockIdx.x * blockDim.x;
-  auto col = input.colids[colStart + blockIdx.y];
+
+  IdxT col;
+  if (input.nSampledCols == input.N) {
+    col = colStart + blockIdx.y;
+  } else {
+    int colIndex = colStart + blockIdx.y;
+    col = select(colIndex, treeid, node.info.unique_id, seed, input.N);
```

Comment:
Nice to see that the feature sampling logic is encapulated in the `select` function.